core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -66,7 +100,8 @@ private[spark] object TaskDescription {
     }
   }
 
-  def encode(taskDescription: TaskDescription): ByteBuffer = {
+  @throws[TaskNotSerializableException]
+  def encode(taskDescription: TaskDescription, serializedTask: ByteBuffer): ByteBuffer = {
```

Comment:
Why do you find it more readable to pass in the serializedTask?  I'd prefer removing serializedTask, since serializedTask is created as part of serializing the taskDescription, which is the functionality that this method is supposed to be doing.  It's confusing to have this encode() method just doing part of the serialization, and expecting the caller to do the other part.